src/tools/ScenarioMeasurement/Startup/WindowsTraceSession.cs

```diff
@@ -0,0 +1,116 @@
+            UserSession.Dispose();
+
+            MergeFiles(KernelSession.FileName, UserSession.FileName, traceFilePath);
+            logger.Log($"Trace Saved to {traceFilePath}");
+        }
+
+        private void MergeFiles(string kernelTraceFile, string userTraceFile, string traceFile)
+        {
+            var files = new List<string>();
+            if (File.Exists(kernelTraceFile))
+            {
+                files.Add(kernelTraceFile);
+            }
+            if (File.Exists(userTraceFile))
+            {
+                files.Add(userTraceFile);
+            }
+            if (files.Count != 0)
+            {
+                logger.Log($"Merging {string.Join(',',files)}... ");
+                TraceEventSession.Merge(files.ToArray(), traceFile);
+                if (File.Exists(traceFile))
+                {
+                    File.Delete(userTraceFile);
+                    File.Delete(kernelTraceFile);
+                }
+            }
+        }
+
+        public void EnableKernelProvider(params TraceSessionManager.KernelKeyword[] keywords)
+        {
+            // Create keyword flags for windows events
+            KernelTraceEventParser.Keywords flags = 0;
+            foreach (var keyword in keywords)
+            {
+                flags |= kernelKeywords[keyword];
+            }
+            KernelSession.EnableKernelProvider(flags);
+        }
+
+        public void EnableUserProvider(params TraceSessionManager.ClrKeyword[] keywords)
+        {
+            // Create keyword flags for windows events
+            ClrPrivateTraceEventParser.Keywords flags = 0;
+            foreach (var keyword in keywords)
+            {
+                flags |= clrKeywords[keyword];
+            }
+            UserSession.EnableProvider(ClrPrivateTraceEventParser.ProviderGuid, TraceEventLevel.Verbose, (ulong)flags);
+        }
+
+        private void InitWindowsKeywordMaps()
+        {
+            // initialize windows kernel keyword map
+            kernelKeywords = new Dictionary<TraceSessionManager.KernelKeyword, KernelTraceEventParser.Keywords>();
+            kernelKeywords[TraceSessionManager.KernelKeyword.Process] = KernelTraceEventParser.Keywords.Process;
+            kernelKeywords[TraceSessionManager.KernelKeyword.Thread] = KernelTraceEventParser.Keywords.Thread;
+            kernelKeywords[TraceSessionManager.KernelKeyword.ContextSwitch] = KernelTraceEventParser.Keywords.ContextSwitch;
+
+            // initialize windows clr keyword map
+            clrKeywords = new Dictionary<TraceSessionManager.ClrKeyword, ClrPrivateTraceEventParser.Keywords>();
+            clrKeywords[TraceSessionManager.ClrKeyword.Startup] = ClrPrivateTraceEventParser.Keywords.Startup;
+        }
+
+        public void EnableUserProvider(string provider)
+        {
+            UserSession.EnableProvider(provider);
+        }
+
+        public string GetTraceFilePath()
+        {
+            return traceFilePath;
+        }
```

Comment:
Why is this a method instead of a property?

Reply:
Misused some Java convention here. Will get rid of this method.